### PR TITLE
Initial Commit.

### PR DESCRIPTION
**Pull request by gibsonbailey**

### CODE REVIEW

File: scripts/GCE_GPU_Startup.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+echo "SUDO_USER =" $SUDO_USER
+
+echo "Adding graphics drivers repo."
+sudo add-apt-repository -y ppa:graphics-drivers/ppa
```
Comment (ckcollab) [inline scripts/GCE_GPU_Startup.sh:5]: Can we remove all these echos? I think if we do something [like this](https://stackoverflow.com/questions/5750450/print-each-command-before-executing) we can not need that?

File: elasticcloud.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+            self.gce.destroy_node(oldest_node)
+            return "Destroyed node: " + oldest_node.name + "."
+        else:
+            return "Only " + self.min_nodes + " nodes running. (min)"
+
+    
+
+
+class ElasticCloud(GCEMixin):
```
Comment (ckcollab) [inline elasticcloud.py]: What's this going to look like when we load in multiple adapters? Below you use `GCEMixin()` even though it's inherited -- I think we should remove this inheritance. 


### Something like this seems to make sense:



```python
ACTION_SHRINK = 'shrink'
ACTION_EXPAND = 'expand'
ACTION_DO_NOTHING = 'do_nothing'

class ElasticCloudAdapter:
    """This class is so we know what an adapter should look like, and if we need any common behavior between them, like loading configuration"""
    def __init__(self):
        self.config = self._load_configuration()
    def _load_configuration(self):
        return dictionary_of_configuration_stuff_somehow
    def expand(self):
        raise NotImplemented()
    def shrink(self):
        raise NotImplemented()
    def get_next_action(self):
        # get status, do we need to shrink or expand? return that


class GCSAdapter(ElasticCloudAdapter):
    def _load_configuration(self):
        super()._load_configuration()  # we can use Super!
        # ... additional tweaks to configuration loading, if needed

   # ... actual implementations of expand, shrink, etc. *specific to GCS*


class AzureAdapter(ElasticCloudAdapter):
   # ... actual implementations of expand, shrink, etc. *specific to Azure*



@click.command()
@click. argument('driver', type=click.Choice(['gcs', 'azure']))
def do_load_check_and_execute_if_needed(driver):
    if driver == "gcs":
       adapter = GCSAdapter()
    # ...
    next_action = adapter.get_next_action()
    if next_action == ACTION_SHRINK:
        adapter.shrink()
    if next_action == ACTION_EXPAND:
        adapter.expand()
```

Then our scheduler runs `do_load_check_and_execute_if_needed gcs` every 5 mins or so! I think that seems nice..

What do you think of that structure? Been a long day, apologies if I'm overlooking something!

**Review by ckcollab — CHANGES_REQUESTED** [review]

File: scripts/GCE_GPU_Startup.sh (hunk 1 above)
Comment (gibsonbailey) [reply scripts/GCE_GPU_Startup.sh:5]: Good call. I'll make that change.

File: elasticcloud.py (hunk 2 above)
Comment (gibsonbailey) [reply elasticcloud.py]: This structure makes much more sense to me now. I'll switch it up to work like this.

**Pull request closed without merging**